libs/common/src/lib/policies/site-polygon.policy.ts

```diff
@@ -4,8 +4,16 @@ import { Op } from "sequelize";
 
 export class SitePolygonPolicy extends UserPermissionsPolicy {
   async addRules() {
+    const user = await this.getUser();
+    const isGreenhouseUser = user?.roles?.some(role => role.name === "greenhouse-service-account") ?? false;
+
     if (this.permissions.includes("polygons-manage")) {
-      this.builder.can("manage", SitePolygon);
+      if (isGreenhouseUser) {
```

Comment:
We want to avoid directly checking roles; the policies should be permissions based as much as possible. In this case, I could check for the `polygons-manage` permission, which is only granted to service accounts.